### PR TITLE
3017.1 fix clippy warnings

### PR DESCRIPTION
**Pull request by roxy-dao**
Part of #3017

# 👩🏻‍💻 What does this PR do? 
Some clippy fixes... there are still a lot.........

# 🧪 How has/should this change been tested? 
Make sure tests are still passing and everything is working as it should

## 💌 Any notes for the reviewer?
Still heaps of clippy changes to be made so this is only part 1!

### CODE REVIEW

File: server/android/src/lib.rs [hunk 1]
```diff
@@ -1 +1,2 @@
+#![allow(clippy::all)]
```
Comment (clemens-msupply) [inline server/android/src/lib.rs:1]: Why did you disable them for Android?

Comment (roxy-dao) [reply server/android/src/lib.rs:1]: Sorry, I just disabled it for now... will enable again when I do the rest of the clippy fixes 👀

Comment (clemens-msupply) [reply server/android/src/lib.rs:1]: ok cool, could you please take it out of the PR though so that we don't forget?

**Review by clemens-msupply — APPROVED** [review]
Haven't looked into the change in detail but tests are still passing

**Pull request merged**